tuxemon/npc.py

```diff
@@ -754,6 +754,38 @@ def remove_monster(self, monster: Monster) -> None:
             self.monsters.remove(monster)
             self.set_party_status()
 
+    def evolve_monster(self, monster: Monster, evolution: str) -> None:
+        """
+        Evolve a monster from this npc's party.
+
+        Parameters:
+            monster: Monster to remove from the npc's party.
+            evolution: Monster to add to the npc's party.
+
+        """
+        if monster in self.monsters:
+            # Store old monster data
+            old_level = monster.level
+            old_current_hp = monster.current_hp
+            old_flairs = monster.flairs
+            self.monsters.remove(monster)
+
+            # TODO: implement an evolution animation
+
+            # Set new monster data
+            new_monster = Monster()
+            new_monster.load_from_db(evolution)
+            new_monster.set_level(old_level)
+            new_monster.current_hp = min(old_current_hp, new_monster.hp)
+            new_monster.slug = new_monster.slug
+            self.monsters.append(new_monster)
```

Comment:
The order has changed :(

Reply:
Because it hasn't maintained the same slot position?

Reply:
Yes

Reply:
There is the slot now, but I need to figured out the right parameters.

Reply:
Because now evolution after battle is fine, now I need to fix the item based.